frontend/packages/console-dynamic-plugin-sdk/src/extensions/pvc.ts

```diff
@@ -0,0 +1,87 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { CodeRef, EncodedCodeRef, UpdateExtensionProperties } from '../types';
+import { ExtensionCommonK8sResource } from '../utils/common';
+
+export namespace ExtensionProperties {
+  export type PVCCreateProp = {
+    /** label for the create prop */
+    label: string;
+    /** path for the create prop */
+    path: string;
+  };
+
+  export type PVCAlert = {
+    /** CodeRef for the corresponding Alert component. */
+    alert: EncodedCodeRef;
+  };
+  export type PVCAlertCodeRefs<T extends ExtensionCommonK8sResource> = {
+    alert: CodeRef<React.FC<{ pvc: T }>>;
```

Comment:
You shouldn't force the use of a function component over a class component in APIs. Allow both.
```suggestion
    alert: CodeRef<React.ComponentType<{ pvc: T }>>;
```